wordpress-mobile/WordPress-Android
# Site creation service

Fixes #6824 

Introduces a Service that performs the site creation steps. A simple "placeholder" UI is used for interacting with the Service and will be expanded to conform to a new design for a site-creation wizard.

The Service uses the also newly introduced `AutoForeground` helper to seamlessly foreground/background the Service. If the app gets backgrounded while the service is still running, a statusbar notification appears to monitor the progress. The notification gets hidden if the app resumes while the service is still running.

This PR targets the dedicated feature branch.

Depends on #6823 to be merged to develop but in the meantime, it already includes the commit needed.

To test:
* Edit your `gradle.properties` file and define the 4 strings needed for the new site creation. See https://github.com/wordpress-mobile/WordPress-Android/blob/86ffba93e0d26d5d08095a5ad5f3cbd4576ffb3b/WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SiteCreatingFragment.java#L82-L85 for the field names required. Feel free to use `intergalactic-2` for the theme name/id.
* Compile the app and run it
* With the app logged in to WPCOM, navigate to MySite, "switch" to a different site and use the "+" button on the site picker's toolbar to start the creation process.
* Feel free to background the app and resume it. Notice the statusbar notification appearing, updating or disappearing.
* When the app UI is in the foreground, the status text below the illustration should update while the process is advancing
* Upon success, the new site should be created, it's tagline set and its theme properly set as well. Use the web to verify.
